src/test/java/com/daml/extensions/testing/SetupAppCallbackWithRestartIT.java

```diff
@@ -28,11 +29,12 @@ public class SetupAppCallbackWithRestartIT {
           .setupAppCallback(client -> boolFlag.set(true))
           .build();
 
-  @ClassRule public static ExternalResource sandboxClassRule = sandbox.getClassRule();
-  @Rule public ExternalResource sandboxRule = sandbox.getRule();
+  public Sandbox getSandbox() {
```

Comment:
Why is this getter needed?

Reply:
we need to restart sandbox after each test inside test class
actually this is an action on SandboxManager object that lives inside Sandbox object
this object is private 
before, with junit4 we used methods that lived in the Sandbox class
and we call them from @Rull and @ClassRule properties
these properties are deleted in junit5
the new approach is to create a separate class that implements these AfterEach methods
these methods takes Test context as a parameter
so what I needed to do is to extract sanboxManager from this context
you can see this code in the new file [SandboxTestExtension.java](https://github.com/digital-asset/lib-daml-jvm-test/pull/81/files#diff-3cca91a6e9d34585b84258325f314adb66d754dade767a7420770e9e61456834)
```
    Object testInstance = context.getRequiredTestInstance();
    Sandbox sandbox =
        (Sandbox) testInstance.getClass().getMethod("getSandbox", null).invoke(testInstance, null);
    sandbox.stop();
```
I can't find a way to extract sandboxManager from context without this getter
May be there is a better approach in java for it... The thing I don't like is that I add this getter in every test class so it is badly duplicated
I tried to add a super class with sanbox protected and this method
but this mean that every test should override sandbox property.
As I learn yesterday overriding properties is not a good idea in java

Reply:
is it not possible to use the property itself (without the getter)?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getField-java.lang.String-